src/librepgp/stream-armor.cpp

```diff
@@ -453,29 +453,35 @@ find_armor_header(const char *buf, size_t len, size_t *hdrlen)
     return NULL;
 }
 
+static bool
+str_equals(const char *str, size_t len, const char *another)
+{
+    return (len >= strlen(another)) && !strncmp(str, another, len);
```

Comment:
@ni4 and you should use `memcmp` here too

Reply:
@rrrooommmaaa Thanks, fixed and re-pushed.